mp_api/client/mprester.py

```diff
@@ -1477,3 +1477,148 @@ def query(*args, **kwargs):
             please see the new documentation.
             """
         )
+
+    def get_cohesive_energy(
+        self,
+        material_id: MPID | str | list[MPID | str],
+        normalization: Literal["atom", "formula_unit"] = "atom",
+    ) -> float | dict[str, float]:
+        """Obtain the cohesive energy of the structure(s) corresponding to one or many MPIDs.
+
+        Args:
+            material_id (MPID or str or [MPID | str]) : a single MPID or a list of many to compute
+                cohesive energies for.
+            normalization (str = "atom" (default) or "formula_unit") :
+                Whether to normalize the cohesive energy by the number of atoms (default)
+                or by the number of formula units.
+                Note that the current default is inconsistent with the legacy API.
+
+        Returns:
+            (float or dict[str,float]) : If only MPID was specified, returns the cohesive energy
+                in eV/atom for that material. If multiple MPIDs were specified, the cohesive
+                energies (in eV/atom) for each material are returned in a dict indexed by MPID.
+        """
+        if isinstance(material_id, MPID | str):
+            material_id = [material_id]
+
+        entry_preference = {
+            k: i for i, k in enumerate(["GGA", "GGA_U", "SCAN", "R2SCAN"])
+        }
+        run_type_to_dfa = {"GGA": "PBE", "GGA_U": "PBE", "R2SCAN": "r2SCAN"}
+
+        energies = {mp_id: {} for mp_id in material_id}
+        entries = self.get_entry_by_material_id(material_id, compatible_only=False)
+        for entry in entries:
+            # Ensure that this works with monty_decode = False and True
+            if isinstance(entry, dict):
```

Comment:
I think this is fine but might be more transparent to use `self.use_document_model` and/or `self.monty_decode` to make the decision

Reply:
Changed to the `monty_decode` check